syntax = "proto3";

package grpc;

service Excel2Csv {
  rpc SayHello (HelloRequest) returns (StringResponse) {}
  rpc UploadFile(stream UploadFileRequest) returns (StringResponse) {}
  rpc DownloadFile(MetaData) returns (stream FileResponse) {}
}

message HelloRequest {
}

message StringResponse {
  string message = 1;
}

message UploadFileRequest {
  oneof request {
    MetaData metadata = 1;
    bytes chunk_data = 2;
  }
}

message MetaData {
  string filename = 1;
  string extension = 2;
}

message FileResponse {
  bytes chunk_data = 1;
}
